lib/ffi/sdr_funcs.go

```diff
@@ -669,3 +678,41 @@ func (sb *SealCalls) TreeD(ctx context.Context, sector storiface.SectorRef, unse
 
 	return nil
 }
+
+func (sb *SealCalls) SyntheticProofs(ctx context.Context, task *harmonytask.TaskID, sector storiface.SectorRef, sealed cid.Cid, unsealed cid.Cid, randomness abi.SealRandomness, pieces []abi.PieceInfo) error {
+	fspaths, pathIDs, releaseSector, err := sb.sectors.AcquireSector(ctx, task, sector, storiface.FTCache|storiface.FTSealed, storiface.FTNone, storiface.PathSealing)
+	if err != nil {
+		return xerrors.Errorf("acquiring sector paths: %w", err)
+	}
+	defer releaseSector()
+
+	ssize, err := sector.ProofType.SectorSize()
+	if err != nil {
+		return err
+	}
+
+	err = ffi.GenerateSynthProofs(sector.ProofType, sealed, unsealed, fspaths.Cache, fspaths.Sealed, sector.ID.Number, sector.ID.Miner, randomness, pieces)
+	if err != nil {
+		return xerrors.Errorf("generating synthetic proof: %w", err)
+	}
+
+	if err = ffi.ClearCache(uint64(ssize), fspaths.Cache); err != nil {
+		return xerrors.Errorf("failed to clear cache for synthetic proof of sector %d of miner %d", sector.ID.Miner, sector.ID.Number)
+	}
+
+	// Generate 3 random commits
+	for i := 0; i < C1CheckNumber; i++ {
+		var sd [32]byte
+		_, _ = rand.Read(sd[:])
+		_, err = ffi.SealCommitPhase1(sector.ProofType, sealed, unsealed, fspaths.Cache, fspaths.Sealed, sector.ID.Number, sector.ID.Miner, randomness, sd[:], pieces)
+		if err != nil {
+			return xerrors.Errorf("checking PreCommit for synthetic proofs for num:%d tkt:%v seed:%v sealedCID:%v, unsealedCID:%v failed: %w", sector.ID.Number, randomness, sd[:], sealed, unsealed, err)
+		}
+	}
```

Comment:
Would be nice to independently also put those check-rounds in TreeRC

Reply:
This would increase the TreeRC time but few seconds if we do this for all sectors. Is that acceptable?

Reply:
Question, does having cache files around would force C1 to use cache files instead of synthetics ones?

Reply:
> This would increase the TreeRC time but few seconds if we do this for all sectors. Is that acceptable?

Yes, that's fine

Reply:
So, this doesn't work for synthetic proof sectors. Synthetic cache files are expected by C1 in this case.

Reply:
I have pushed a new commit with required changes.